src/celeritas/optical/ScintillationParams.cc

```diff
@@ -30,66 +30,68 @@ std::shared_ptr<ScintillationParams>
 ScintillationParams::from_import(ImportData const& data,
                                  SPConstParticles particle_params)
 {
-    CELER_EXPECT(!data.optical.empty());
+    CELER_EXPECT(!data.optical_materials.empty());
 
-    if (!std::any_of(
-            data.optical.begin(), data.optical.end(), [](auto const& iter) {
-                return static_cast<bool>(iter.second.scintillation);
-            }))
+    if (!std::any_of(data.optical_materials.begin(),
+                     data.optical_materials.end(),
+                     [](auto const& iter) {
+                         return static_cast<bool>(iter.scintillation);
+                     }))
     {
         // No scintillation data present
         return nullptr;
     }
 
-    auto const& num_optmats = data.optical.size();
+    size_type const num_optmats = data.optical_materials.size();
 
     Input input;
     input.resolution_scale.resize(num_optmats);
     if (data.optical_params.scintillation_by_particle)
     {
         // Collect ScintillationParticleIds
         input.pid_to_scintpid.resize(data.particles.size());
-        ScintillationParticleId scintpid{0};
-        for (auto const& [matid, iom] : data.optical)
+        ScintillationParticleId::size_type scintpid{0};
```

Comment:
Just so I am in the same page, let me know if this statement is correct: the plan is to keep the scintillation by particle import code, but we're not implementing that yet in our generator. So we keep per-particle data on the import side (and `ScintillationData`, though will be empty), but do not enable it in our `ScintillationParams`, leaving the stepping loop to be material-only. And if/when we do add scintillation by particle type, all the backend and data storage is already in place, since the current extra bits shouldn't make any real difference in the stepping loop.

Reply:
Yeah, that's about the sum of it. Since I don't think we've yet found any sincillation-by-particle detector we could probably delete it and restore it later if needed... 🤔